clients/command-kafka/src/main/java/org/eclipse/hono/client/command/kafka/KafkaBasedCommandContext.java

```diff
@@ -81,8 +91,20 @@ public void release(final Throwable error) {
             return;
         }
         final Span span = getTracingSpan();
-        TracingHelper.logError(span, "command for device handled with outcome 'released'");
-        span.finish();
+        if (isRequestResponseCommandWithCorrelationId() && !(error instanceof CommandAlreadyProcessedException)) {
+            final ServiceInvocationException mappedError = StatusCodeMapper.toServerError(error);
+            final String errorMessage = Optional.ofNullable(ServiceInvocationException.getErrorMessageForExternalClient(mappedError))
+                    .orElse("Temporarily unavailable");
+            sendDeliveryFailureCommandResponseMessage(mappedError.getErrorCode(), errorMessage, span)
+                    .onComplete(v -> {
+                        TracingHelper.logError(span, "command for device handled with outcome 'released'");
+                        span.finish();
+                    });
+        } else {
+            logErrorIfCorrelationIdMissing(span);
+            TracingHelper.logError(span, "command for device handled with outcome 'released'");
```

Comment:
shouldn't this also include the error object?

Reply:
Currently the exception already gets traced in the places the `released`/`rejected` method gets invoked. 
I've created #2730 to adapt the invocations and improve the tracing. After that PR gets merged, I'll rebase this PR here.

Reply:
PR is rebased now.